### PR TITLE
DLO-18 (2)

### PR DESCRIPTION
**Pull request by genli-se761**
### **Related Issue/Keyword:**

https://github.com/Willning/DLOplanTool/pull/11 continuation.

Fixed an error that was caused by the changes.

### **Description:**

Firestore.js and Firestore.test.js contain the Firestore class that contains functions for interfacing with the Firestore database and corresponding tests respectively.

### **Testing:**

Execute **npm test** within the root directory.

### **Checklist:**

- [x] Latest master merged/rebased into your feature branch
- [ ] Meets the projects coding conventions
- [X] No out of scope changes
- [X] #Mentioned other relevant issues
- [X] No failure when running the application (obviously lol)

### CODE REVIEW

**Comment by genli-se761** [comment]
> ah, sorry, I didn't realize the bug still exists before because there are lots of test data in firestore, but I cleared the test data a momment ago, when running npm test, everything goes well, when running npm start, there are 8 user documents inserted into firestore and not been deleted

Hi Huizhou, I wasn't able to reproduce the issue and so guess it might be related to local development environment differences. Could you please try and see if you can reproduce it consistently?

**Comment by joe-zhan** [comment]
> > ah, sorry, I didn't realize the bug still exists before because there are lots of test data in firestore, but I cleared the test data a momment ago, when running npm test, everything goes well, when running npm start, there are 8 user documents inserted into firestore and not been deleted
> 
> Hi Huizhou, I wasn't able to reproduce the issue and so guess it might be related to local development environment differences. Could you please try and see if you can reproduce it consistently?

Sorry, wrong info provided. nothing about npm start, just npm test, when you stared at the firestore console during running npm test, you will see green and red states of documents seems everything goes well, everytime a doc will be deleted after insert, however if you refresh the firestore console page after test finished, you will see some test records that not been deleted and also not shown realtime on the firestore console page during test.

File: src/Firestore.js [hunk 1]
```diff
@@ -6,41 +6,39 @@ var db = app.firestore();
...
-        var docRef = db.collection(collection).doc(docID);
-        docRef.get().then(function (doc) {
-            if (doc.exists) {
-                return doc.data();
-            } else {
-                console.log("No such document!");
-            }
-        }).catch(function (error) {
-            console.log("Error getting document:", error);
+    static saveUser(email, username) {
```
Comment (hybrio) [inline src/Firestore.js]: please update this to have the same fields as the updateUserDetails function

File: src/Firestore.js [hunk 2]
```diff
@@ -52,30 +50,26 @@ class Firestore {
...
+    static saveProjectToUser(userID, project) {
+        return this.saveToDBWithDocID(
+            this.getAllProjectsByUser(userID),
+            project.id,
+            project
+        );
     };
 
     static saveNewProject(userID, projectData) {
-        this.updateUserDetails();
```
Comment (hybrio) [inline src/Firestore.js]: by removing this updateUserDetails is no longer called (meaning that the user's details are never saved to the db) either re-add the function call somewhere else or don't remove it.

**Pull request merged**